lib/davinci_dtr_test_kit/client_groups/dinner_adaptive/dtr_adaptive_questionnaire_next_question_request_validation_test.rb

```diff
@@ -0,0 +1,64 @@
+require_relative '../../urls'
+
+module DaVinciDTRTestKit
+  class DTRAdaptiveQuestionnaireNextQuestionRequestValidationTest < Inferno::Test
+    include URLs
+
+    id :dtr_next_question_request_validation
+    title '$next-question request is valid'
+    description %(
+      Per the [OperationDefinition: Adaptive questionnaire next question](https://build.fhir.org/ig/HL7/sdc/OperationDefinition-Questionnaire-next-question.html#root)
+      section in the [Structured Data Capture IG](http://hl7.org/fhir/uv/sdc/ImplementationGuide/hl7.fhir.uv.sdc),
+      the request body for the `$next-question` operation should be a FHIR Parameters resource containing a
+      single parameter with:
+        - name: `questionnaire-response`
+        - resource: A `QuestionnaireResponse` resource
+
+      As outlined in the [FHIR Operation Request](https://hl7.org/fhir/r4/operations.html#request) section of the
+      FHIR specification, if an operation has exactly one input parameter of type Resource, it can also be invoked via
+      a POST request using that resource as the body (with no additional URL parameters).
+
+      This test validates the structure of the `$next-question` request body. It confirms that the body is either a
+      Parameters resource or a QuestionnaireResponse resource.
+
+      If it is a Parameters resource, it must contain one parameter named `questionnaire-response`
+      with a `resource` attribute set to a FHIR QuestionnaireResponse instance, as specified above.
+      Additionally, this test checks that the Parameters resource conforms to the [SDC Next Question Input Parameters](http://hl7.org/fhir/uv/sdc/StructureDefinition/parameters-questionnaire-next-question-in)
+      profile.
+
+      If the request body is a QuestionnaireResponse resource, its structure and conformance will be validated
+      in the following test ('Adaptive QuestionnaireResponse is valid').
+    )
+
+    def assert_valid_resource_type(resource)
+      type = resource.resourceType
+      valid = ['Parameters', 'QuestionnaireResponse'].include?(type)
+      assert valid, "Request body not valid. Expected Parameters or QuestionnaireResponse, got #{type}"
+    end
+
+    run do
+      load_tagged_requests CLIENT_NEXT_TAG
+      skip_if request.blank?, 'A $next-question request must be made prior to running this test'
+
+      assert request.url == next_url, "Request made to wrong URL: #{request.url}. Should instead be to #{next_url}"
+      assert_valid_json(request.request_body)
+      input_params = FHIR.from_contents(request.request_body)
+      assert input_params.present?, 'Request does not contain a recognized FHIR object'
+      assert_valid_resource_type(input_params)
+
+      if input_params.is_a?(FHIR::Parameters)
+        assert_valid_resource(
+          resource: input_params,
+          profile_url: 'http://hl7.org/fhir/uv/sdc/StructureDefinition/parameters-questionnaire-next-question-in'
```

Comment:
I don't think we want to do this profile validation. 
1. I suspect no validation will actually be done because the parameter name used in "in" rather than "questionnaire-response" which we and the spec use everywhere else.
2. Even if validation is done, it will just duplicate stuff in the next test if it goes down deeper into the QuestionnaireResponse.

So let's get rid of it.